Services.Test/InternalDevicePropertiesTest.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Services.Test.helpers;
+using Xunit;
+using Xunit.Abstractions;
+using static Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.DeviceModel;
+
+namespace Services.Test
+{
+    public class InternalDevicePropertiesTest
+    {
+        private InternalDeviceProperties target;
+
+        public InternalDevicePropertiesTest(ITestOutputHelper log)
+        {
+            // Initialize device properties
+            this.target = this.GetTestProperties();
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void GetWithNoPropertiesIsEmpty()
+        {
+            // Arrange
+            this.target = this.GetEmptyProperties();
+
+            // Act
+            var props = this.target.GetAll();
+
+            // Assert
+            Assert.Empty(props);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void GetPropertiesIsNotNullAndHasExpectedCount()
+        {
+            // Arrange
+            this.target = this.GetTestProperties();
+            var expectedCount = this.GetTestChillerModel().Properties.Count;
+
+            // Act
+            var props = this.target.GetAll();
+
+            // Assert
+            Assert.NotEmpty(props);
+            Assert.Equal(props.Count, expectedCount);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void GetPropertyThrowsIfKeyDoesNotExist()
+        {
+            // Arrange
+            this.target = this.GetTestProperties();
+            var key = "KeyThatDoesNotExist";
+
+            // Act and Assert
+            Assert.Throws<KeyNotFoundException>(() => this.target.Get(key));
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void GetPropertyReturnsValue()
+        {
+            // Arrange
+            this.target = this.GetTestProperties();
+            var key = "TestPropKey";
+            var value = "TestPropValue";
+
+            // Act
+            var result = this.target.Get(key);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(result, value);
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void SetPropertyTest()
```

Comment:
There are two new classes added. Upon discussion, we decided it was best for properties not to inherit from DeviceState In case device state behavior changes in the future and breaks the properties logic for checking if a value is "dirty" and needs to be updated to the twin. 

So yes, they are the same, but it was intentional.